INT-1275 - Add a check for errors returned in XML response body

mknoedel opened the pull request.

aiwilliams reviewed (dismissed):
Very nice! Love that this will reflect other codes as well to the user so they might address the issue.

ndowmon commented on src/provider/client/index.ts:
```diff
@@ -227,6 +228,16 @@ export class QualysAPIClient {
+    const jsonFromXml = xmlParser.parse(responseText) as
+      | string
+      | was.ActivityLogErrorResponse;
+    if (_.isObject(jsonFromXml)) {
```
Just want to point this out - This seems to imply that any time this response is JSON, it means something errored. Is that a safe assumption or too broad?

mknoedel replied:
Good catch. I added a check for content type to be more explicit now.

ndowmon commented on src/provider/client.test.ts:
```diff
@@ -436,6 +436,18 @@ describe('verifyAuthentication', () => {
+      options: { recordFailedRequests: true },
+    });
+
+    await expect(createClient().verifyAuthentication()).rejects.toBeInstanceOf(
```
Not a big deal, but I like to see the actual error message that would be thrown in tests like this.

mknoedel replied:
Added

ndowmon reviewed (commented):
Looks good! I just added a comment that I wanted to point out, plus a small suggestion (you can choose to implement the suggestion or not)

mknoedel commented:
> Looks good! I just added a comment that I wanted to point out, plus a small suggestion (you can choose to implement the suggestion or not)

Great suggestions Nick thanks! Updated
